### PR TITLE
Daniel/single column histogram

### PR DESCRIPTION
**Pull request by dandanlen**
For Issue #6

### CODE REVIEW

File: src/explorer.api/Queries/SingleColumnHistogram.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+        public string ColumnName { get; }
+
+        public IList<decimal> Buckets { get; }
+
+        public class Result : IJsonArrayConvertible
+        {
+            public decimal? BucketSize { get; set; }
+
+            public AircloakColumn<decimal> LowerBound { get; set; }
```
Comment (AndreiBozantan) [inline src/explorer.api/Queries/SingleColumnHistogram.cs]: The compiler issues a warning here regarding the uninitialized non-nullable field.

File: src/explorer.api/Queries/DistinctColumnValues.cs [hunk 2]
```diff
@@ -34,7 +35,7 @@ public DistinctColumnValues(string tableName, string columnName)
 
         public class IntegerResult : IJsonArrayConvertible
         {
-            public long? ColumnValue { get; set; }
+            public AircloakColumn<long> ColumnValue { get; set; }
```
Comment (AndreiBozantan) [inline src/explorer.api/Queries/DistinctColumnValues.cs]: The compiler issues a warning here regarding the uninitialized non-nullable field (this is the same for all result types).

File: src/aircloak/JsonApi/ResponseTypes/Columns/AircloakColumn.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+namespace Aircloak.JsonApi.ResponseTypes
+{
+    /// <summary>
+    /// Abstract base class for Diffix column values which can be suppressed or NULL.
+    /// </summary>
+    /// <typeparam name="T">The expected type of the column value.</typeparam>
+    public abstract class AircloakColumn<T>
```
Comment (AndreiBozantan) [inline src/aircloak/JsonApi/ResponseTypes/Columns/AircloakColumn.cs]: The generic parameter is not used inside the class. Could it be removed and added just in the required derived class?

File: src/explorer.api/Queries/SingleColumnHistogram.cs (hunk 1 above)
Comment (dandanlen) [reply src/explorer.api/Queries/SingleColumnHistogram.cs]: Fixed this.

File: src/explorer.api/Queries/DistinctColumnValues.cs (hunk 2 above)
Comment (dandanlen) [reply src/explorer.api/Queries/DistinctColumnValues.cs]: ☝️ Fixed

File: src/aircloak/JsonApi/ResponseTypes/Columns/AircloakColumn.cs (hunk 3 above)
Comment (dandanlen) [reply src/aircloak/JsonApi/ResponseTypes/Columns/AircloakColumn.cs]: I wasn't sure about this - it's true the parameter isn't used in the abstract class however if we keep it, it ensures that the columns are always tagged with the expected return type. If a column value is suppressed or null, that doesn't mean we lose the type information - so it should be useful to keep this and allow the type-checker to ensure that we don't mix columns of different types. What do you think?

**Pull request merged**